packages/grpc-js/src/server.ts

```diff
@@ -1388,52 +1541,123 @@ export class Server {
           }
         }, this.maxConnectionAgeMs + jitter).unref?.();
       }
+
       const keeapliveTimeTimer: NodeJS.Timeout | null = setInterval(() => {
         const timeoutTImer = setTimeout(() => {
           sessionClosedByServer = true;
-          if (this.channelzEnabled) {
-            this.channelzTrace.addTrace(
-              'CT_INFO',
-              'Connection dropped by keepalive timeout from ' + clientAddress
-            );
-          }
+          this.channelzTrace.addTrace(
+            'CT_INFO',
+            'Connection dropped by keepalive timeout from ' + clientAddress
+          );
+
           session.close();
         }, this.keepaliveTimeoutMs).unref?.();
         try {
           session.ping(
             (err: Error | null, duration: number, payload: Buffer) => {
               clearTimeout(timeoutTImer);
+
+              if (err) {
+                sessionClosedByServer = true;
+                this.channelzTrace.addTrace(
+                  'CT_INFO',
+                  `Connection dropped due to error of a ping frame ${err.message} return in ${duration}`
```

Comment:
Similar to the regular trace logs, I think we should avoid string interpolation in channelz trace logs, especially now that the string will always be created even if channelz is disabled.